projects/plugins/jetpack/modules/subscriptions/subscribe-overlay/class-jetpack-subscribe-overlay.php

```diff
@@ -0,0 +1,229 @@
+		if ( apply_filters( 'jetpack_subscribe_overlay_enabled', false ) ) {
+			add_action( 'wp_enqueue_scripts', array( $this, 'enqueue_assets' ) );
+			add_action( 'wp_footer', array( $this, 'add_subscribe_overlay_to_frontend' ) );
+		}
+
+		add_filter( 'get_block_template', array( $this, 'get_block_template_filter' ), 10, 3 );
+	}
+
+	/**
+	 * Returns the block template part ID.
+	 *
+	 * @return string
+	 */
+	public static function get_block_template_part_id() {
+		return get_stylesheet() . '//' . self::BLOCK_TEMPLATE_PART_SLUG;
+	}
+
+	/**
+	 * Makes get_block_template return the WP_Block_Template for the Subscribe Overlay.
+	 *
+	 * @param WP_Block_Template $block_template The block template to be returned.
+	 * @param string            $id Template unique identifier (example: theme_slug//template_slug).
+	 * @param string            $template_type Template type: `'wp_template'` or '`wp_template_part'`.
+	 *
+	 * @return WP_Block_Template|null
+	 */
+	public function get_block_template_filter( $block_template, $id, $template_type ) {
+		if ( empty( $block_template ) && $template_type === 'wp_template_part' ) {
+			if ( $id === self::get_block_template_part_id() ) {
+				return $this->get_template();
+			}
+		}
+
+		return $block_template;
+	}
+
+	/**
+	 * Returns a custom template for the Subscribe Overlay.
+	 *
+	 * @return WP_Block_Template
+	 */
+	public function get_template() {
+		$template                 = new WP_Block_Template();
+		$template->theme          = get_stylesheet();
+		$template->slug           = self::BLOCK_TEMPLATE_PART_SLUG;
+		$template->id             = self::get_block_template_part_id();
+		$template->area           = 'uncategorized';
+		$template->content        = $this->get_subscribe_overlay_template_content();
+		$template->source         = 'plugin';
+		$template->type           = 'wp_template_part';
+		$template->title          = __( 'Jetpack Subscribe overlay', 'jetpack' );
+		$template->status         = 'publish';
+		$template->has_theme_file = false;
+		$template->is_custom      = true;
+		$template->description    = __( 'An overlay that shows up when someone visits your site', 'jetpack' );
+
+		return $template;
+	}
+
+	/**
+	 * Returns the initial content of the Subscribe Overlay template.
+	 * This can then be edited by the user.
+	 *
+	 * @return string
+	 */
+	public function get_subscribe_overlay_template_content() {
+		$site_name        = get_bloginfo( 'name' );
+		$site_description = get_bloginfo( 'description' ); // TODO can be empty, how to handle?
+		$skip_to_content  = __( 'Skip to content', 'jetpack' );
+
+		return <<<HTML
+	<!-- wp:group {"layout":{"type":"constrained","contentSize":"400px"}} -->
+	<div class="wp-block-group">
+		<!-- wp:site-logo {"width":90,"shouldSyncIcon":true,"align":"center","className":"is-style-rounded"} /-->
+	
+		<!-- wp:heading {"textAlign":"center","level":1} -->
+		<h1 class="wp-block-heading has-text-align-center">$site_name</h1>
+		<!-- /wp:heading -->
```

Comment:
Would be better to use "Site title" block here:

<img width="344" alt="Screenshot 2024-05-14 at 14 28 50" src="https://github.com/Automattic/jetpack/assets/87168/b2707b80-5b83-4324-aef8-cc0db5269436">

Reply:
fixed, thanks!